src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(state, { ...obj.extraData });
+        break;
+
+      case 'removeProperties':
+        const deleteData = obj.keysToRemove;
+
+        for (const deleteKey of deleteData) {
+          delete state[deleteKey];
+        }
+        break;
+
+      default:
```

Comment:
it's better to do nothing or throw an error on `default` case
```suggestion
      case 'clear':
```